tests/Doctrine/Tests/ORM/Functional/AdvancedDqlQueryTest.php

```diff
@@ -93,6 +93,48 @@ public function testOrderByResultVariableCollectionSize()
         $this->assertEquals(1, $result[3]['friends']);
     }
 
+    public function testOrderBySimpleCaseExpression() : void
+    {
+        $dql = 'SELECT p.name ' .
+               'FROM Doctrine\Tests\Models\Company\CompanyEmployee p ' .
+               'ORDER BY CASE p.name ' .
+               'WHEN \'Jonathan W.\' THEN 1 ' .
+               'WHEN \'Roman B.\' THEN 2 ' .
+               'WHEN \'Guilherme B.\' THEN 3 ' .
+               'ELSE 4 ' .
+               'END DESC';
```

Comment:
Let's avoid the big amount of concatenation and escaping:

```suggestion
        $dql = <<<'DQL'
        SELECT p.name
          FROM Doctrine\Tests\Models\Company\CompanyEmployee p
      ORDER BY CASE p.name
          WHEN 'Jonathan W.' THEN 1
          WHEN 'Roman B.' THEN 2
          WHEN 'Guilherme B.' THEN 3
          ELSE 4
          END DESC
DQL;
```